### PR TITLE
Add Alice & Lucidia cooperative agent framework (provide models package)

### PR DESCRIPTION
**Pull request by blackboxprogramming**
## Summary
- add the alice_lucidia package implementing the Alice planner and Lucidia world-model agents with Hopfield memory and OT utilities
- provide training, evaluation, CLI, and notebook assets plus configuration and packaging metadata for the cooperative agent system
- cover critical behaviours with pytest suites for Hopfield energy, OT routines, natural-gradient optimiser, and agent planning logic
- include the alice_lucidia.models package with encoder, Hopfield, world model, natural gradient, and OT primitives so Lucidia imports succeed
- allow the models package to bypass the repository ignore rules

## Testing
- pytest -q alice_lucidia/tests

------
https://chatgpt.com/codex/tasks/task_e_68fa94f9bbd48329a0a7317437d23400

### CODE REVIEW

File: alice_lucidia/models/hopfield.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        proj_memory = self.key_proj(candidates)
+        scale = math.sqrt(self.config.projection_dim)
+        scores = torch.einsum("bd,bmd->bm", proj_query, proj_memory) / scale
+        weights = torch.softmax(self.config.beta * scores, dim=-1)
+        readout = torch.einsum("bm,bmd->bd", weights, candidates)
+        shift = scores.max(dim=-1, keepdim=True).values
+        stabilised = self.config.beta * (scores - shift)
+        logsum = torch.logsumexp(stabilised, dim=-1)
+        energy = -(shift.squeeze(-1) + logsum / max(self.config.beta, 1e-6))
+        return readout, energy
```
[the remark this answers is not in the document]
Comment (blackboxprogramming) [reply alice_lucidia/models/hopfield.py]: @copilot @dependabot @blackroad @cadillac @codex @lucidia @cecilia @gitguardian @blackboxprogramming @slackbot @linear @slack @asanabot @blackroadagents @blackroadbots @airtablebot review and fix comments please!
